packages/gatsby/src/commands/develop.ts

```diff
@@ -286,21 +286,22 @@ module.exports = async (program: IProgram): Promise<void> => {
       process.send(msg)
     }
 
+    io.emit(`structured-log`, msg)
+
     if (
       msg.type === `LOG_ACTION` &&
       msg.action.type === `SET_STATUS` &&
       msg.action.payload === `SUCCESS`
     ) {
       proxy.serveSite()
-      io.emit(`develop:started`)
     }
   }
 
   io.on(`connection`, socket => {
-    socket.on(`develop:restart`, async () => {
+    socket.on(`develop:restart`, async ackFn => {
       isRestarting = true
       proxy.serveRestartingScreen()
-      io.emit(`develop:is-starting`)
+      ackFn(`develop:is-starting`)
```

Comment:
Could you explain this? I'm not super familiar with socket.io and unsure what ackFn does here. Does it send a new message? Maybe worth a comment in code?

Reply:
It sends back an acknowledgement, kinda like a response to a HTTP request! On the client, this allows us to listen for it like so:

```JS
socket.emit(`develop:restart`, (answer) => {
  console.log(answer) // => would be "develop:is-starting" here
})
```

I'll add a comment 👍

Reply:
Added a comment _and_ renamed the variable to `respond` for clarity 👍